src/context/librariesContext.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, Dispatch, useEffect, useState } from 'react';
+import { LibraryData } from '../components/RegistryAdmin';
+
+export type LibrariesContextValues = {
+  librariesInContext: LibraryData[];
+  setLibrariesInContext: Dispatch<React.SetStateAction<LibraryData[]>>;
+  setUpdatedLibrary: Dispatch<React.SetStateAction<LibraryData | undefined>>;
+};
+
+type LibrariesProviderProps = {
+  children: React.ReactNode;
+};
+
+export const LibrariesContext = createContext<LibrariesContextValues | null>(
+  null
```

Comment:
Let's initialize these context values with `undefined` instead of `null` -- this will make some of the types easier to deal with.

Then, we can wrap this context inside of a function called `useLibraries` which will check for undefined and throw an error. Something like this can be added to the bottom of this file:

```
export default function useLibraries() {
  const context = React.useContext(LibrariesContext);
  if (typeof context === "undefined") {
    throw new Error("useLibraries must be used within a LibraryProvider");
  }
  return context;
}
```

Then, instead of passing the context name into the component, you can instead use:

```
const { librariesInContext } = useLibraries();
```

in the `LibrariesList.tsx` component for example.

We should use this same pattern with the token context too. This also eliminates the need to use the type coercion `as LibrariesContextValues` when you're using the context value.

Reply:
This is the blog post where I discovered this pattern and it has served me well thus far: https://kentcdodds.com/blog/how-to-use-react-context-effectively

Reply:
Cool – this exactly describes my issue and the solution  😂. Thank you!